apps/main/src/lend/hooks/useUserLoanDetails.ts

```diff
@@ -0,0 +1,28 @@
+import useStore from '@/lend/store/useStore'
+import type { HealthColorKey, UserLoanDetails } from '@/lend/types/lend.types'
+
+type Details = UserLoanDetails['details'] & { error: string }
+
+/**
+ * Retrieves user loan details for a specific user active key.
+ * @param userActiveKey - The unique identifier for the user's active loan
+ * @returns User loan details object with error property merged for easier destructuring
+ */
+export function useUserLoanDetails(userActiveKey: string): Details {
+  const loanDetails = useStore((state) => state.user.loansDetailsMapper[userActiveKey])
+
+  return !loanDetails || loanDetails.details == null
+    ? ({} as Details)
```

Comment:
this is a bad cast, if the details can be empty we should change the type to have optional fields